data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessor.java

```diff
@@ -95,6 +96,9 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
             final List<Event> concludeGroupEvents = actionOutput != null ? actionOutput.getEvents() : null;
             if (!concludeGroupEvents.isEmpty()) {
                 concludeGroupEvents.stream().forEach((event) -> {
+                    if (aggregatedEventsTag != null) {
+                        event.getMetadata().addTags(List.of(aggregatedEventsTag));
```

Comment:
It might be nice to add an `addTag(String)` method to EventMetadata to simplify life for the callers, but this isn't critical.